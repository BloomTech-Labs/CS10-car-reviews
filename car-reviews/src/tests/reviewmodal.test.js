import ReviewModal from '../components/Modals/reviewmodal';

describe('Review Modal', () => {
  const props = {
    title: 'The best car ever!',
    content: 'Seriously a great car!',
    score: 5,
    carImage: '',
    car: {
      year: 2018,
      make: 'Toyota',
      model: 'Camry'
    },
    user: {
      username: 'robinhood'
    }
  };

  const wrapper = shallow(<ReviewModal {...props} />);

  // All tests will go here
  it('shallow renders without crashing', () => {
    wrapper;
    expect(wrapper).toMatchSnapshot();
  });

  it('always renders a div', () => {
    const divs = wrapper.find('div');
    expect(divs.length).toBeGreaterThan(0);
  });

  describe('the rendered div', () => {
    it('contains everything else that gets rendered', () => {
      const divs = wrapper.find('div');
      const wrappingDiv = divs.first();

      expect(wrappingDiv.children()).toEqual(wrapper.children());
    });
  });
  describe('Testing state of Review Modal', () => {
    it('modal is defaulted to false', () => {
      expect(wrapper.state('modal')).toBe(false);
    });

    it('counter is defaulted to 0', () => {
      expect(wrapper.state('counter')).toBe(0);
    });

    it('paid flag for a user is defaulted to false before updateUserCounter is called', () => {
      expect(wrapper.state('paid')).toBe(false);
    });
  });
});
