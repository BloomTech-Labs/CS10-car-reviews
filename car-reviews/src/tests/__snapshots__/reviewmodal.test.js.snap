// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Modal shallow renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReviewModal
    car={
        Object {
            "make": "Toyota",
            "model": "Camry",
            "year": 2018,
          }
    }
    carImage=""
    content="Seriously a great car!"
    score={5}
    title="The best car ever!"
    user={
        Object {
            "username": "robinhood",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          className="modal-button"
          color="secondary"
          onClick={[Function]}
          tag="button"
>
          <p
                    style={
                              Object {
                                        "fontSize": "1.1em",
                                      }
                    }
          >
                    2018 Toyota Camry
          </p>
          <div
                    style={
                              Object {
                                        "height": "150px",
                                      }
                    }
          >
                    <img
                              alt=""
                              src="default_img.png"
                              style={
                                        Object {
                                                  "maxHeight": "100%",
                                                  "maxWidth": "100%",
                                                }
                              }
                    />
          </div>
          <div
                    style={
                              Object {
                                        "display": "flex",
                                        "justifyContent": "center",
                                      }
                    }
          >
                    <ReactStars
                              char="★"
                              color1="gray"
                              color2="#ffd700"
                              count={5}
                              edit={false}
                              half={true}
                              name="score"
                              onChange={[Function]}
                              size={30}
                              type="number"
                              value={5}
                    />
          </div>
          <p
                    style={
                              Object {
                                        "fontSize": ".8em",
                                      }
                    }
          >
                    Review by: robinhood
          </p>
</Button>,
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
                    Object {
                              "mountOnEnter": true,
                              "timeout": 150,
                            }
          }
          centered={false}
          fade={true}
          isOpen={false}
          keyboard={true}
          modalTransition={
                    Object {
                              "timeout": 300,
                            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          role="dialog"
          toggle={[Function]}
          zIndex={1050}
>
          <ModalHeader
                    charCode={215}
                    className="modal-header"
                    closeAriaLabel="Close"
                    tag="h5"
                    toggle={[Function]}
                    wrapTag="div"
          >
                    <p
                              style={
                                        Object {
                                                  "fontSize": "1.7em",
                                                }
                              }
                    >
                              2018 Toyota Camry
                    </p>
                    <ReactStars
                              char="★"
                              color1="gray"
                              color2="#ffd700"
                              count={5}
                              edit={false}
                              half={true}
                              name="score"
                              onChange={[Function]}
                              size={30}
                              type="number"
                              value={5}
                    />
          </ModalHeader>
          <ModalBody
                    className="modal-body"
                    tag="div"
          >
                    <img
                              alt=""
                              src="default_img.png"
                              style={
                                        Object {
                                                  "height": "50%",
                                                  "width": "50%",
                                                }
                              }
                    />
                    <p>
                              The best car ever!
                    </p>
                    <p>
                              Seriously a great car!
                    </p>
          </ModalBody>
</Modal>,
      ],
      "className": "button-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <p
              style={
                            Object {
                                          "fontSize": "1.1em",
                                        }
              }
>
              2018 Toyota Camry
</p>,
            <div
              style={
                            Object {
                                          "height": "150px",
                                        }
              }
>
              <img
                            alt=""
                            src="default_img.png"
                            style={
                                          Object {
                                                        "maxHeight": "100%",
                                                        "maxWidth": "100%",
                                                      }
                            }
              />
</div>,
            <div
              style={
                            Object {
                                          "display": "flex",
                                          "justifyContent": "center",
                                        }
              }
>
              <ReactStars
                            char="★"
                            color1="gray"
                            color2="#ffd700"
                            count={5}
                            edit={false}
                            half={true}
                            name="score"
                            onChange={[Function]}
                            size={30}
                            type="number"
                            value={5}
              />
</div>,
            <p
              style={
                            Object {
                                          "fontSize": ".8em",
                                        }
              }
>
              Review by: robinhood
</p>,
          ],
          "className": "modal-button",
          "color": "secondary",
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2018 Toyota Camry",
              "style": Object {
                "fontSize": "1.1em",
              },
            },
            "ref": null,
            "rendered": "2018 Toyota Camry",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                src="default_img.png"
                style={
                                Object {
                                                "maxHeight": "100%",
                                                "maxWidth": "100%",
                                              }
                }
/>,
              "style": Object {
                "height": "150px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "default_img.png",
                "style": Object {
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ReactStars
                char="★"
                color1="gray"
                color2="#ffd700"
                count={5}
                edit={false}
                half={true}
                name="score"
                onChange={[Function]}
                size={30}
                type="number"
                value={5}
/>,
              "style": Object {
                "display": "flex",
                "justifyContent": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "char": "★",
                "color1": "gray",
                "color2": "#ffd700",
                "count": 5,
                "edit": false,
                "half": true,
                "name": "score",
                "onChange": [Function],
                "size": 30,
                "type": "number",
                "value": 5,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Review by: robinhood",
              "style": Object {
                "fontSize": ".8em",
              },
            },
            "ref": null,
            "rendered": "Review by: robinhood",
            "type": "p",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "backdrop": true,
          "backdropTransition": Object {
            "mountOnEnter": true,
            "timeout": 150,
          },
          "centered": false,
          "children": Array [
            <ModalHeader
              charCode={215}
              className="modal-header"
              closeAriaLabel="Close"
              tag="h5"
              toggle={[Function]}
              wrapTag="div"
>
              <p
                            style={
                                          Object {
                                                        "fontSize": "1.7em",
                                                      }
                            }
              >
                            2018 Toyota Camry
              </p>
              <ReactStars
                            char="★"
                            color1="gray"
                            color2="#ffd700"
                            count={5}
                            edit={false}
                            half={true}
                            name="score"
                            onChange={[Function]}
                            size={30}
                            type="number"
                            value={5}
              />
</ModalHeader>,
            <ModalBody
              className="modal-body"
              tag="div"
>
              <img
                            alt=""
                            src="default_img.png"
                            style={
                                          Object {
                                                        "height": "50%",
                                                        "width": "50%",
                                                      }
                            }
              />
              <p>
                            The best car ever!
              </p>
              <p>
                            Seriously a great car!
              </p>
</ModalBody>,
          ],
          "fade": true,
          "isOpen": false,
          "keyboard": true,
          "modalTransition": Object {
            "timeout": 300,
          },
          "onClosed": [Function],
          "onOpened": [Function],
          "role": "dialog",
          "toggle": [Function],
          "zIndex": 1050,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "charCode": 215,
              "children": Array [
                <p
                  style={
                                    Object {
                                                      "fontSize": "1.7em",
                                                    }
                  }
>
                  2018 Toyota Camry
</p>,
                <ReactStars
                  char="★"
                  color1="gray"
                  color2="#ffd700"
                  count={5}
                  edit={false}
                  half={true}
                  name="score"
                  onChange={[Function]}
                  size={30}
                  type="number"
                  value={5}
/>,
              ],
              "className": "modal-header",
              "closeAriaLabel": "Close",
              "tag": "h5",
              "toggle": [Function],
              "wrapTag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2018 Toyota Camry",
                  "style": Object {
                    "fontSize": "1.7em",
                  },
                },
                "ref": null,
                "rendered": "2018 Toyota Camry",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "char": "★",
                  "color1": "gray",
                  "color2": "#ffd700",
                  "count": 5,
                  "edit": false,
                  "half": true,
                  "name": "score",
                  "onChange": [Function],
                  "size": 30,
                  "type": "number",
                  "value": 5,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  src="default_img.png"
                  style={
                                    Object {
                                                      "height": "50%",
                                                      "width": "50%",
                                                    }
                  }
/>,
                <p>
                  The best car ever!
</p>,
                <p>
                  Seriously a great car!
</p>,
              ],
              "className": "modal-body",
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "default_img.png",
                  "style": Object {
                    "height": "50%",
                    "width": "50%",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "The best car ever!",
                },
                "ref": null,
                "rendered": "The best car ever!",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Seriously a great car!",
                },
                "ref": null,
                "rendered": "Seriously a great car!",
                "type": "p",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            className="modal-button"
            color="secondary"
            onClick={[Function]}
            tag="button"
>
            <p
                        style={
                                    Object {
                                                "fontSize": "1.1em",
                                              }
                        }
            >
                        2018 Toyota Camry
            </p>
            <div
                        style={
                                    Object {
                                                "height": "150px",
                                              }
                        }
            >
                        <img
                                    alt=""
                                    src="default_img.png"
                                    style={
                                                Object {
                                                            "maxHeight": "100%",
                                                            "maxWidth": "100%",
                                                          }
                                    }
                        />
            </div>
            <div
                        style={
                                    Object {
                                                "display": "flex",
                                                "justifyContent": "center",
                                              }
                        }
            >
                        <ReactStars
                                    char="★"
                                    color1="gray"
                                    color2="#ffd700"
                                    count={5}
                                    edit={false}
                                    half={true}
                                    name="score"
                                    onChange={[Function]}
                                    size={30}
                                    type="number"
                                    value={5}
                        />
            </div>
            <p
                        style={
                                    Object {
                                                "fontSize": ".8em",
                                              }
                        }
            >
                        Review by: robinhood
            </p>
</Button>,
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropTransition={
                        Object {
                                    "mountOnEnter": true,
                                    "timeout": 150,
                                  }
            }
            centered={false}
            fade={true}
            isOpen={false}
            keyboard={true}
            modalTransition={
                        Object {
                                    "timeout": 300,
                                  }
            }
            onClosed={[Function]}
            onOpened={[Function]}
            role="dialog"
            toggle={[Function]}
            zIndex={1050}
>
            <ModalHeader
                        charCode={215}
                        className="modal-header"
                        closeAriaLabel="Close"
                        tag="h5"
                        toggle={[Function]}
                        wrapTag="div"
            >
                        <p
                                    style={
                                                Object {
                                                            "fontSize": "1.7em",
                                                          }
                                    }
                        >
                                    2018 Toyota Camry
                        </p>
                        <ReactStars
                                    char="★"
                                    color1="gray"
                                    color2="#ffd700"
                                    count={5}
                                    edit={false}
                                    half={true}
                                    name="score"
                                    onChange={[Function]}
                                    size={30}
                                    type="number"
                                    value={5}
                        />
            </ModalHeader>
            <ModalBody
                        className="modal-body"
                        tag="div"
            >
                        <img
                                    alt=""
                                    src="default_img.png"
                                    style={
                                                Object {
                                                            "height": "50%",
                                                            "width": "50%",
                                                          }
                                    }
                        />
                        <p>
                                    The best car ever!
                        </p>
                        <p>
                                    Seriously a great car!
                        </p>
            </ModalBody>
</Modal>,
        ],
        "className": "button-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <p
                style={
                                Object {
                                                "fontSize": "1.1em",
                                              }
                }
>
                2018 Toyota Camry
</p>,
              <div
                style={
                                Object {
                                                "height": "150px",
                                              }
                }
>
                <img
                                alt=""
                                src="default_img.png"
                                style={
                                                Object {
                                                                "maxHeight": "100%",
                                                                "maxWidth": "100%",
                                                              }
                                }
                />
</div>,
              <div
                style={
                                Object {
                                                "display": "flex",
                                                "justifyContent": "center",
                                              }
                }
>
                <ReactStars
                                char="★"
                                color1="gray"
                                color2="#ffd700"
                                count={5}
                                edit={false}
                                half={true}
                                name="score"
                                onChange={[Function]}
                                size={30}
                                type="number"
                                value={5}
                />
</div>,
              <p
                style={
                                Object {
                                                "fontSize": ".8em",
                                              }
                }
>
                Review by: robinhood
</p>,
            ],
            "className": "modal-button",
            "color": "secondary",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2018 Toyota Camry",
                "style": Object {
                  "fontSize": "1.1em",
                },
              },
              "ref": null,
              "rendered": "2018 Toyota Camry",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  src="default_img.png"
                  style={
                                    Object {
                                                      "maxHeight": "100%",
                                                      "maxWidth": "100%",
                                                    }
                  }
/>,
                "style": Object {
                  "height": "150px",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "default_img.png",
                  "style": Object {
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ReactStars
                  char="★"
                  color1="gray"
                  color2="#ffd700"
                  count={5}
                  edit={false}
                  half={true}
                  name="score"
                  onChange={[Function]}
                  size={30}
                  type="number"
                  value={5}
/>,
                "style": Object {
                  "display": "flex",
                  "justifyContent": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "char": "★",
                  "color1": "gray",
                  "color2": "#ffd700",
                  "count": 5,
                  "edit": false,
                  "half": true,
                  "name": "score",
                  "onChange": [Function],
                  "size": 30,
                  "type": "number",
                  "value": 5,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Review by: robinhood",
                "style": Object {
                  "fontSize": ".8em",
                },
              },
              "ref": null,
              "rendered": "Review by: robinhood",
              "type": "p",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": true,
            "backdrop": true,
            "backdropTransition": Object {
              "mountOnEnter": true,
              "timeout": 150,
            },
            "centered": false,
            "children": Array [
              <ModalHeader
                charCode={215}
                className="modal-header"
                closeAriaLabel="Close"
                tag="h5"
                toggle={[Function]}
                wrapTag="div"
>
                <p
                                style={
                                                Object {
                                                                "fontSize": "1.7em",
                                                              }
                                }
                >
                                2018 Toyota Camry
                </p>
                <ReactStars
                                char="★"
                                color1="gray"
                                color2="#ffd700"
                                count={5}
                                edit={false}
                                half={true}
                                name="score"
                                onChange={[Function]}
                                size={30}
                                type="number"
                                value={5}
                />
</ModalHeader>,
              <ModalBody
                className="modal-body"
                tag="div"
>
                <img
                                alt=""
                                src="default_img.png"
                                style={
                                                Object {
                                                                "height": "50%",
                                                                "width": "50%",
                                                              }
                                }
                />
                <p>
                                The best car ever!
                </p>
                <p>
                                Seriously a great car!
                </p>
</ModalBody>,
            ],
            "fade": true,
            "isOpen": false,
            "keyboard": true,
            "modalTransition": Object {
              "timeout": 300,
            },
            "onClosed": [Function],
            "onOpened": [Function],
            "role": "dialog",
            "toggle": [Function],
            "zIndex": 1050,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "charCode": 215,
                "children": Array [
                  <p
                    style={
                                        Object {
                                                            "fontSize": "1.7em",
                                                          }
                    }
>
                    2018 Toyota Camry
</p>,
                  <ReactStars
                    char="★"
                    color1="gray"
                    color2="#ffd700"
                    count={5}
                    edit={false}
                    half={true}
                    name="score"
                    onChange={[Function]}
                    size={30}
                    type="number"
                    value={5}
/>,
                ],
                "className": "modal-header",
                "closeAriaLabel": "Close",
                "tag": "h5",
                "toggle": [Function],
                "wrapTag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2018 Toyota Camry",
                    "style": Object {
                      "fontSize": "1.7em",
                    },
                  },
                  "ref": null,
                  "rendered": "2018 Toyota Camry",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "char": "★",
                    "color1": "gray",
                    "color2": "#ffd700",
                    "count": 5,
                    "edit": false,
                    "half": true,
                    "name": "score",
                    "onChange": [Function],
                    "size": 30,
                    "type": "number",
                    "value": 5,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <img
                    alt=""
                    src="default_img.png"
                    style={
                                        Object {
                                                            "height": "50%",
                                                            "width": "50%",
                                                          }
                    }
/>,
                  <p>
                    The best car ever!
</p>,
                  <p>
                    Seriously a great car!
</p>,
                ],
                "className": "modal-body",
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "default_img.png",
                    "style": Object {
                      "height": "50%",
                      "width": "50%",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "The best car ever!",
                  },
                  "ref": null,
                  "rendered": "The best car ever!",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Seriously a great car!",
                  },
                  "ref": null,
                  "rendered": "Seriously a great car!",
                  "type": "p",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
